tests/Set-DbaStartupParameter.Tests.ps1

```diff
@@ -4,16 +4,53 @@ Write-Host -Object "Running $PSCommandPath" -ForegroundColor Cyan
 
 Describe "$CommandName Unit Tests" -Tag 'UnitTests' {
     Context "Validate parameters" {
-        [object[]]$params = (Get-Command $CommandName).Parameters.Keys | Where-Object {$_ -notin ('whatif', 'confirm')}
+        [array]$params = ([Management.Automation.CommandMetaData]$ExecutionContext.SessionState.InvokeCommand.GetCommand($CommandName, 'Function')).Parameters.Keys
         [object[]]$knownParameters = 'SqlInstance', 'SqlCredential', 'Credential', 'MasterData', 'MasterLog', 'ErrorLog', 'TraceFlag', 'CommandPromptStart', 'MinimalStart', 'MemoryToReserve', 'SingleUser', 'SingleUserDetails', 'NoLoggingToWinEvents', 'StartAsNamedInstance', 'DisableMonitoring', 'IncreasedExtents', 'TraceFlagOverride', 'StartUpConfig', 'Offline', 'Force', 'EnableException'
-        $knownParameters += [System.Management.Automation.PSCmdlet]::CommonParameters
         It "Should only contain our specific parameters" {
-            (@(Compare-Object -ReferenceObject ($knownParameters | Where-Object {$_}) -DifferenceObject $params).Count ) | Should Be 0
+            Compare-Object -ReferenceObject $knownParameters -DifferenceObject $params | Should -BeNullOrEmpty
         }
     }
 }
 <#
     Integration test should appear below and are custom to the command you are writing.
     Read https://github.com/sqlcollaborative/dbatools/blob/development/contributing.md#tests
     for more guidence.
-#>
\ No newline at end of file
+#>
+
+Describe "$CommandName Integration Tests" -Tags "IntegrationTests" {
+
+    BeforeAll {
+        $defaultInstance = $script:instance1
+        $namedInstance = $script:instance2
+    }
+
+    Context "Bug 7035" {
+        # Adding the -Skip here to ensure the AppVeyor instance isn't modified. This test can be run locally.
+        It -Skip "Ensure the startup params are not duplicated when more than one server is modified in the same invocation" {
+            $result = Set-DbaStartupParameter -SqlInstance $defaultInstance, $namedInstance -TraceFlag 3226 -Confirm:$false
+
+            $resultDefaultInstance = Get-DbaStartupParameter -SqlInstance $defaultInstance
+            $resultDefaultInstance.TraceFlags.Count | Should -Be 1
+            $resultDefaultInstance.TraceFlags[0] | Should -Be 3226
+
+            # The duplication occurs after the first server is processed.
+            $resultNamedInstance = Get-DbaStartupParameter -SqlInstance $namedInstance
+            # Using the defaults to test locally
+            $resultNamedInstance.MasterData.Count | Should -Be 1
+            $resultNamedInstance.MasterLog.Count | Should -Be 1
+            $resultNamedInstance.ErrorLog.Count | Should -Be 1
+
+            $resultNamedInstance.TraceFlags.Count | Should -Be 1
+            $resultNamedInstance.TraceFlags[0] | Should -Be 3226
+        }
+
+        # Adding the -Skip here to ensure the AppVeyor instance isn't modified. This test can be run locally.
+        It -Skip "Ensure the correct instance name is returned" {
```

Comment:
I don't know that we want to get into using Skip for tests. A normal user is not going to even notice these without reviewing the physical tests, and it can't be run unless you remove the Skip parameter. Pester does not offer any mechanism to force skip tests to run.

If we have tests like this that can only be run locally then I would say we need to implement a Tag for it. @niphlod ?

I mean we could start adding validation to see if we are in the Appveyor environment, but that functionality will be broken when we move to Pester 5+. The discovery process won't make that a viable option, the tagging though would.

If we used the tag then we just need to have our base test run to exclude that tag for local-only tests.

Reply:
It's possible to use -Skip:$false and -Skip:$true to control whether a test would run. So, perhaps with a $script variable defined in constants.ps1 it could provide the on/off switch for running local-only tests. There are several dbatools commands that have pester tests with the -Skip param, so they could be updated to use that variable.

Reply:
we just have tests/pester.groups.ps1 to identify an entire function in the "appveyor_exclude" stanza. I think we shouldn't be "this conservative" with appveyor, and running tests on a local instance SHOULD ASSUME that the environment is not pristine afterwards. 
Skip is usually just implemented to avoid running tests where the setup failed.
I suggest to make this run within the service_restarts env (same file to let this happen, tests/pester.groups.ps1), which is where "really bad things" happen.

Reply:
Moved it to the `service_restarts` group.